crates/mpz-ot-core/src/chou_orlandi/receiver.rs

```diff
@@ -140,10 +145,18 @@ impl Receiver<state::Setup> {
     /// * `payload` - The encrypted payload from the Sender
     pub fn receive(&mut self, payload: SenderPayload) -> Result<Vec<Block>, ReceiverError> {
         let state::Setup {
-            decryption_keys, ..
+            transfer_id: current_id,
+            decryption_keys,
+            ..
         } = &mut self.state;
 
-        let SenderPayload { payload } = payload;
+        let SenderPayload { id, payload } = payload;
+
+        // Check that the transfer id matches
+        let expected_id = current_id.next();
+        if id != expected_id {
+            return Err(ReceiverError::IdMismatch(id, expected_id));
```

Comment:
This is in the wrong order.
```suggestion
            return Err(ReceiverError::IdMismatch(expected_id, id));
```